components/raftstore/src/store/fsm/store.rs

```diff
@@ -2436,6 +2445,65 @@ impl<'a, EK: KvEngine, ER: RaftEngine, T: Transport> StoreFsmDelegate<'a, EK, ER
         }
     }
 
+    fn register_full_compact_tick(&self) {
+        if !self.ctx.cfg.periodic_full_compact_start_times.is_empty() {
+            self.ctx.schedule_store_tick(
+                StoreTick::PeriodicFullCompact,
+                PERIODIC_FULL_COMPACT_TICK_INTERVAL_DURATION,
+            )
+        }
+    }
+
+    fn on_full_compact_tick(&mut self) {
+        self.register_full_compact_tick();
+
+        let local_time = chrono::Local::now();
+        if !self
+            .ctx
+            .cfg
+            .periodic_full_compact_start_times
+            .is_scheduled_this_hour(&local_time)
+        {
+            debug!(
+                "full compaction may not run at this time";
+                "local_time" => ?local_time,
+                "periodic_full_compact_start_times" => ?self.ctx.cfg.periodic_full_compact_start_times,
+            );
+            return;
+        }
+
+        if self.ctx.global_stat.stat.is_busy.load(Ordering::SeqCst) {
```

Comment:
I don't think we need lower memory order to have better performance here (since this won't happen frequently), but just out of curiosity, do we need SeqCst here?

Reply:
We do not. I believe our rust guide tells us to use SeqCst unless we do:
```
7. Unless in the performance critical section, use SeqCst for atomic operations.
```
from our internal wiki (TiKV coding guidelines.)